lucene/suggest/src/java/org/apache/lucene/search/suggest/analyzing/BlendedInfixSuggester.java

```diff
@@ -200,7 +201,13 @@ protected FieldType getTextFieldType() {
       textDV.advance(fd.doc);
 
       final String text = textDV.binaryValue().utf8ToString();
-      long weight = (Long) fd.fields[0];
+
+      NumericDocValues weightDV = MultiDocValues.getNumericValues(searcher.getIndexReader(), WEIGHT_FIELD_NAME);
```

Comment:
Could you please make local variables final to prevent reassignment.

Reply:
Thank you for your note, It was not the main scope of the bug fix, but It's a good recommendation so I just contributed the little refactor.
Thanks!

Reply:
Thanks for incorporating the changes!!!!!